core/src/test/scala/com/itv/scheduler/Generators.scala

```diff
@@ -0,0 +1,28 @@
+package com.itv.scheduler
+
+import cats.Eq
+import org.scalacheck._
+import org.scalacheck.ScalacheckShapeless._
+import org.scalacheck.derive._
+
+trait Generators {
+  // Not quite sure how to define these any other way ...
+  implicit def eqJobDecoder[A]: Eq[JobDecoder[A]]         = Eq.allEqual[JobDecoder[A]]
+  implicit def eqJobDataEncoder[A]: Eq[JobDataEncoder[A]] = Eq.allEqual[JobDataEncoder[A]]
+  implicit def eqJobCodec[A]: Eq[JobCodec[A]]             = Eq.allEqual[JobCodec[A]]
```

Comment:
This doesn't seem great to me, but I don't know what else to define them as. Help me Obi Wan Jack-Wheatley 🙏

Reply:
well, 2 `JobDecoder`s are equal iff their read functions are equal. So should be I guess
```scala
implicit def eqJobDecoder[A](implicit ev: Eq[(Chain[String], PartiallyDecodedJobData) => Either[Throwable, A]]) = 
    Eq.by[JobDecoder[A], (Chain[String], PartiallyDecodedJobData) => Either[Throwable, A]](_.read)
```

etc

Reply:
so think these are the most "correct'

```scala
implicit def eqJobDecoder[A](implicit ev: Eq[(Chain[String], PartiallyDecodedJobData) => Either[Throwable, A]]): Eq[JobDecoder[A]] =
    Eq.by[JobDecoder[A], (Chain[String], PartiallyDecodedJobData) => Either[Throwable, A]](_.read)
  implicit def eqJobDataEncoder[A](implicit ev: Eq[A => JobData]): Eq[JobDataEncoder[A]] = Eq.by[JobDataEncoder[A], A => JobData](_.encode)
  implicit def eqJobCodec[A](implicit ev1: Eq[JobDecoder[A]], ev2: Eq[JobDataEncoder[A]]): Eq[JobCodec[A]] = Eq.and(ev1, ev2)
```

Reply:
Oh good point, thanks @jbwheatley - I'll try taking a look to see if I can use the functions